### ISSUE TITLE
JodaTime throws exception on hour shifting

### ISSUE THREAD

**Issue by danilo-valente** [jmrozanec/cron-utils#19]
This code:

```java
CronDefinition definition = CronDefinitionBuilder.instanceDefinitionFor(CronType.QUARTZ);
CronParser parser = new CronParser(definition);
Cron cron = parser.parse("0/1 * * 1/1 * ? *");  // Every 1 second
ExecutionTime executionTime = ExecutionTime.forCron(cron);

DateTime now = new DateTime().withTime(23, 59, 59, 0);
DateTime nextExecution = executionTime.nextExecution(now);
```

Would throw this exception: `org.joda.time.IllegalFieldValueException: Value 24 for hourOfDay must be in the range [0,23]` because 23:59:59 + 1 second would become 24:00:00, which is an invalid hour for a new `DateTime` object.

**Comment by jmrozanec** [comment]
Good catch! Thank you for contributing the fix!

**Comment by danilo-valente** [comment]
@jmrozanec Glad to help! Do you have any idea on when v1.1.3 is going to be released?

**Comment by jmrozanec** [comment]
@danilo-valente I will be releasing it this weekend.

**Comment by danilo-valente** [comment]
@jmrozanec Thanks :smile:

**Comment by jmrozanec** [comment]
@danilo-valente you are welcome!

**Comment by jmrozanec** [comment]
@danilo-valente new version released! New version number is 2.0.0, since there were some incompatible API changes.
